### ISSUE TITLE
Use Ollama for transformers

### ISSUE THREAD

**Issue by mclevey** [mclevey/podlm#24]
Use [Ollama](https://ollama.ai) for transformer models wherever it makes sense to do so. 

Ollama is 🔥🔥🔥

**Comment by tcrick** [comment]
Agree! But need to update my knowledge of the recently flooded landscape.

My first selection criterion to dig into for choosing a base model will be the maximum sequence length, which isn't necessarily as easy as looking at the maximum that the sequence length parameter will accept at model inference. See discussion: https://www.reddit.com/r/LocalLLaMA/comments/13ed7re/4096_context_length_and_beyond/

Most of the current generation of models are built on a much smaller selection of cores - GPT, Llama, etc. The length of the training sequences for the core models matters but I don't know how much. 

Strictly speaking, from the most thorough answer in that thread, this is probably correct: "Llama isn't trained on more than 2K token sequences. It simply doesn't know what to do if a state vector more than 2k spaces into the past encodes a past of its own, because that's not a situation it's ever encountered during training. The result is degenerate output, even if you truncate the past key/value outputs to try a sliding-window approach hoping to carry over information that would be lost if you had truncated the input sequence instead"

More loosely speaking, from another response: "The paper for ALiBi suggests that the model generalizes quite well to about 2-3x the trained context length without even finetuning. So 4k trained model should handle about 10k before it gets noticeably confused. "

This also seems to be a quite knowledgeable answer. It will be important to know how accurate the 2-3x figure is. "degenerate output" could mean it degenerates by only 1-2% at the start of what I assume will be an exponential curve. I also suspect that how much this degeneracy matters will be dependent on our inference task and context.

**Why this matters:** the more we have to chop up reddit posts (or other documents) into chunks, the more we have to come up with - and justify - some way to aggregate the inference results from those chunks if we want document-level analysis. 

**Happy to hear if true**: longer sequences mean a quadratic increase to complexity for model training but closer to a linear complexity increase for inference.

**Unsure**: what longer sequences mean for VRAM requirements. Probably just smaller batch sizes.

**Comment by tcrick** [comment]
Labeling this as "hard" because the LLM of choice will be fundamental to many of the other processes, so I will want to be very clear and certain about it.
